src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -81,7 +82,14 @@ public class RSessionProvider : IRSessionProvider {
                 return;
             }
 
-            foreach (var session in _sessions.Values) {
+            var sessions = GetSessions();
+            var stopHostTasks = sessions.Select(session => session.StopHostAsync());
+            try {
+                Task.WhenAll(stopHostTasks).GetAwaiter().GetResult();
```

Comment:
In this case, shouldn't we schedule it to run in UI thread?

---
In reply to: [82472969](https://github.com/Microsoft/RTVS/pull/2465#discussion_r82472969) [](ancestors = 82472969)